river_dl/train.py

```diff
@@ -21,33 +21,102 @@ def get_data_if_file(d):
         return np.load(d, allow_pickle=True)
 
 
+# This is a training engine that initializes our model and contains routines for pretraining and finetuning
+class trainer():
+    def __init__(self, model, optimizer, loss_fn, weights=None):
+        self.model = model
+        self.model.compile(optimizer=optimizer, loss=loss_fn)
+        if weights:
+            self.model.load_weights(weights)
+
+    def pre_train(self, x, y, epochs, batch_size, out_dir):
+
+        ## Set up training log callback
+        csv_log = tf.keras.callbacks.CSVLogger(
+            os.path.join(out_dir, f"pretrain_log.csv")
+        )
+
+        # Use generic fit statement
+        self.model.fit(
+            x=x,
+            y=y,
+            epochs=epochs,
+            batch_size=batch_size,
+            callbacks=[csv_log],
+        )
+        # Save the pretrained weights
+        self.model.save_weights(os.path.join(out_dir, "pretrained_weights/"))
+        return self.model
+
+    def fine_tune(self, x, y, x_val, y_val, epochs, batch_size, out_dir, early_stop_rounds=False):
+        # Specify our training log
+        csv_log = tf.keras.callbacks.CSVLogger(
+            os.path.join(out_dir, "finetune_log.csv")
+        )
+
+        # Set up early stopping rounds if desired, setting this to number of epochs functionally doesn't use
+        # early stopping
+        if ~early_stop_rounds:
```

Comment:
I think this would be clearer and do the same thing (and maybe be more conventional):
```suggestion
        if not early_stop_rounds:
```